[Source Defect]: e-lindsey

### Source Name

e_lindsey_gov_uk

### What's happening?

update_docu_links is flagging an issue with e-lindsey source:

```bash
Orphaned sources without country =========================
filename:/doc/source/e_lindsey_gov_uk.md, title:East Lindsey District Council, url:https://www.e-lindsey.gov.uk, country:gb, params:['uprn'], extra_info_default_params:{}, cusom_param_translation:{'de': {'uprn': 'UPRN'}, 'en': {'uprn': 'UPRN'}, 'fr': {'uprn': 'UPRN'}, 'it': {'uprn': 'UPRN'}}
```
Shouldn't the country code be "uk"  rather than "gb"?

### Error Logs

```Shell
see above
```

### Source Configuration

```YAML
see above
```

### Checklist

- [x] I have tested with the latest version (master branch) via HACS → 3-dot menu → Redownload → select `master`
- [x] The service provider's website is still working
- [x] I have tested with the example parameters from the documentation
- [x] I have searched past issues (open and closed) for this problem

I was going to submit a quick PR with the change, but testing the script is giving errors for the same file:

```bash
Testing source e_lindsey_gov_uk ...
  13 Firbeck Avenue, Skegness failed: We could not find values for the argument 'uprn' with the value '100030786099', Could not find form prefix on page
    Traceback (most recent call last):
      File "/home/d/Repos/hacs_waste_collection_schedule_v2/custom_components/waste_collection_schedule/waste_collection_schedule/test/test_sources.py", line 135, in test_fetch
        result = source.fetch()
                 ^^^^^^^^^^^^^^
      File "/home/d/Repos/hacs_waste_collection_schedule_v2/custom_components/waste_collection_schedule/waste_collection_schedule/source/e_lindsey_gov_uk.py", line 78, in fetch
        raise SourceArgumentNotFound(
    waste_collection_schedule.exceptions.SourceArgumentNotFound: We could not find values for the argument 'uprn' with the value '100030786099', Could not find form prefix on page
```

The UPRN seems valid, and matches the address, so something else must be amiss.

update_docu_links also suggests there are some other issues :

```bash
> git -c user.useConfigOnly=true commit --quiet --allow-empty-message --file -
pyupgrade................................................................Passed
black....................................................................Passed
codespell................................................................Failed
- hook id: codespell
- exit code: 65

README.md:2434: Aare ==> Are
info.md:41: Aare ==> Are

flake8...................................................................Passed
bandit...................................................................Passed
Format YAML files....................................(no files to check)Skipped
isort....................................................................Passed
check json...............................................................Passed
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

custom_components/waste_collection_schedule/waste_collection_schedule/source/e_lindsey_gov_uk.py: note: In member "fetch" of class "Source":
custom_components/waste_collection_schedule/waste_collection_schedule/source/e_lindsey_gov_uk.py:98: error:
Item "None" of "Match[str] | None" has no attribute "group"  [union-attr]
                f"{prefix}_pageSessionId": page_session_id.group(1),
                                           ^~~~~~~~~~~~~~~~~~~~~
custom_components/waste_collection_schedule/waste_collection_schedule/source/e_lindsey_gov_uk.py:99: error:
Item "None" of "Match[str] | None" has no attribute "group"  [union-attr]
                f"{prefix}_fsid": fsid.group(1),
                                  ^~~~~~~~~~
custom_components/waste_collection_schedule/waste_collection_schedule/source/e_lindsey_gov_uk.py:100: error:
Item "None" of "Match[str] | None" has no attribute "group"  [union-attr]
                f"{prefix}_fsn": fsn.group(1),
                                 ^~~~~~~~~
Found 3 errors in 1 file (checked 1 source file)

```

The codespell error is unrelated, see #6071.

--- a/custom_components/waste_collection_schedule/waste_collection_schedule/source/e_lindsey_gov_uk.py
+++ b/custom_components/waste_collection_schedule/waste_collection_schedule/source/e_lindsey_gov_uk.py
@@ -14,15 +14,6 @@ TEST_CASES = {
     "13 Firbeck Avenue, Skegness": {"uprn": "100030786099"},
 }
 
-EXTRA_INFO = [
-    {
-        "title": TITLE,
-        "url": URL,
-        "country": "gb",
-        "default_params": {},
-    }
-]
-
 HOW_TO_GET_ARGUMENTS_DESCRIPTION = {
     "en": "Find your UPRN using a tool like findmyaddress.co.uk or finders.io."
 }
@@ -38,7 +29,6 @@ ICON_MAP = {
 }
 
 _PAGE_URL = "https://www.e-lindsey.gov.uk/mywastecollections"
-_SUBMIT_URL = "https://www.e-lindsey.gov.uk/mywastecollections/processsubmission"
 
 _ORDINAL_RE = re.compile(r"(\d+)(st|nd|rd|th)")
 
@@ -47,9 +37,8 @@ def _parse_date(s: str) -> datetime.date | None:
     """Parse e.g. 'Wednesday 29th April 2026' → date."""
     if not s:
         return None
-    s = _ORDINAL_RE.sub(r"\1", s)  # strip ordinal suffix
+    s = _ORDINAL_RE.sub(r"\1", s)
     parts = s.split()
-    # format: "Weekday DD Month YYYY"
     if len(parts) == 4:
         try:
             return datetime.datetime.strptime(
@@ -73,42 +62,75 @@ class Source:
         html = r.text
 
         # Extract form prefix (e.g. WASTECOLLECTIONDAYS202627)
-        prefix_match = re.search(r'name="([A-Z0-9]+)_pageSessionId"', html)
+        prefix_match = re.search(r'name="([A-Z0-9]+)_PAGESESSIONID"', html)
         if not prefix_match:
             raise SourceArgumentNotFound(
                 "uprn", self._uprn, "Could not find form prefix on page"
             )
         prefix = prefix_match.group(1)
 
-        # Extract session tokens
-        page_session_id = re.search(
-            rf'name="{prefix}_pageSessionId"\s+value="([^"]+)"', html
+        # Extract form action URL (contains pageSessionId, fsid, fsn as query params)
+        action_match = re.search(
+            r'<form[^>]+action="(https://[^"]+/processsubmission[^"]*)"', html
         )
-        fsid = re.search(rf'name="{prefix}_fsid"\s+value="([^"]+)"', html)
-        fsn = re.search(rf'name="{prefix}_fsn"\s+value="([^"]+)"', html)
+        if not action_match:
+            raise SourceArgumentNotFound(
+                "uprn", self._uprn, "Could not find form action URL on page"
+            )
+        submit_url = action_match.group(1).replace("&amp;", "&")
 
-        if not all([page_session_id, fsid, fsn]):
+        # Extract hidden session tokens
+        pagesid_match = re.search(
+            rf'name="{prefix}_PAGESESSIONID"\s+value="([^"]+)"', html
+        )
+        sessionid_match = re.search(
+            rf'name="{prefix}_SESSIONID"\s+value="([^"]+)"', html
+        )
+        nonce_match = re.search(rf'name="{prefix}_NONCE"\s+value="([^"]+)"', html)
+
+        if not pagesid_match or not sessionid_match or not nonce_match:
             raise SourceArgumentNotFound(
                 "uprn", self._uprn, "Could not extract form session tokens"
             )
 
-        # Step 2: POST to processsubmission with UPRN
-        variables = json.dumps({"UPRN": self._uprn})
+        # Step 2: POST to processsubmission with UPRN in base64-encoded VARIABLES
+        # ADDRESSSOURCE must be NLPGEL (National Land and Property Gazetteer for East Lindsey)
+        variables = base64.b64encode(
+            json.dumps(
+                {
+                    "ADDRESSSOURCE": {
+                        "value": "NLPGEL",
+                        "scope": "SERVERCLIENTWITHUPDATE",
+                    },
+                    "ADDRESSUPRN": {
+                        "value": self._uprn,
+                        "scope": "SERVERCLIENTWITHUPDATE",
+                    },
+                    "TESTDATELAYOUT_DISPLAYED": {
+                        "value": False,
+                        "scope": "SERVERCLIENT",
+                    },
+                }
+            ).encode()
+        ).decode()
+
         form_data = {
-            f"{prefix}_pageSessionId": page_session_id.group(1),
-            f"{prefix}_fsid": fsid.group(1),
-            f"{prefix}_fsn": fsn.group(1),
-            f"{prefix}_FIELD1": self._uprn,
+            f"{prefix}_PAGESESSIONID": pagesid_match.group(1),
+            f"{prefix}_SESSIONID": sessionid_match.group(1),
+            f"{prefix}_NONCE": nonce_match.group(1),
             f"{prefix}_VARIABLES": variables,
-            f"{prefix}_submit": "1",
+            f"{prefix}_PAGENAME": "LOOKUP",
+            f"{prefix}_PAGEINSTANCE": "0",
+            f"{prefix}_LOOKUP_CHOSENADDRESS": self._uprn,
+            f"{prefix}_FORMACTION_NEXT": f"{prefix}_LOOKUP_FIELD2",
         }
 
-        r = session.post(_SUBMIT_URL, data=form_data, timeout=30)
+        r = session.post(submit_url, data=form_data, timeout=30)
         r.raise_for_status()
 
-        # Step 3: Extract base64-encoded FIELD12 data from response JS
+        # Step 3: Parse FormData from the results page
         data_match = re.search(
-            rf'var\s+{prefix}_RESULTS_FIELD12Data\s*=\s*"([^"]+)"',
+            rf"{prefix}FormData\s*=\s*\"([^\"]+)\"",
             r.text,
         )
         if not data_match:
@@ -118,8 +140,7 @@ class Source:
                 "No collection data found — check that the UPRN is valid",
             )
 
-        decoded = base64.b64decode(data_match.group(1)).decode("utf-8")
-        payload = json.loads(decoded)
+        payload = json.loads(base64.b64decode(data_match.group(1)).decode("utf-8"))
 
         result_list = payload.get("RESULTS_1", {}).get("FIELD12", {}).get("result", [])
         if not result_list:

